Revert "Upgrade Node and Npm version for edxapp (#6716)"

aht007 opened the pull request:
Reverts openedx/configuration#6758

nedbat commented:
I see a number of revert pull requests. Can you provide some context about why they are being reverted?

aht007 commented:
These changes broke the nutmeg release because Tutor uses `npm ci` and with our changes we had moved to `npm install`. You can see the whole conversation [here](https://openedx.slack.com/archives/C01AGTSB1LL/p1653409070765259) about the possibilities and why we choose to keep `edx-platform` on Node12 for Nutmeg and reverted the changes
